### DISCUSSION TITLE
Extension deactivate method on browser reload scenario.

### DISCUSSION THREAD

**Discussion by AxelFlores1990** [coder/code-server#5615] [General]
Hi! how are you?

I find some issue when the deactivate method of extension was executed on browser tab reload..

The deactivate was executed, but in standalone version, vscode provide 5 secs of grace period([as metion here](https://github.com/Microsoft/vscode/issues/47881#issuecomment-381910587)) to stop every process before shutdown the host process..

I think that in web browser this 5 seconds was not provided as grace period..

We use the activate/deactivate methods to run a daemon process that works as a runtime to provide some services.. and in the standalone version it stop fine, but in the web browser the stop process was interrupted drastically...

Do you know if this is an issue or if theres any other way to map the reload tab event from the extension side?

**Comment by jsjoeio** [comment]
This is an interesting question! Do you know if this happens in Codespaces? If so, it's worth mentioning upstream. If not, then it's in code-server and we should probably fix it.

Maybe @code-asher has some insight.

**Reply by code-asher** [reply]
Yeah this sounds likely to be an upstream issue.  I think in web it just terminates the extension host without a grace period.  Related: https://github.com/coder/code-server/issues/2882

**Reply by jsjoeio** [reply]
@AxelFlores1990 see here^

**Reply by AxelFlores1990** [reply]
I think its not the same, in my case the deactivate method was called..

**Reply by AxelFlores1990** [reply]
Sorry, could you explain with a little bit more of detail the scenario? I don't understand who is responsible of the extension host kill without the grace period...

**Reply by jsjoeio** [reply]
>  I don't understand who is responsible of the extension host kill without the grace period..

I believe this happens on the VS Code side.

**Reply by code-asher** [reply]
Yeah that issue is not exactly the same, I just linked it because it is somewhat related (web shutdown logic being different from standalone).  Like @jsjoeio said this happens in VS Code.  code-server wraps VS Code for the web so all of that logic is on their side.  Sometimes we do add patches for bugs in code-server but we try to send them upstream when we can.

**Reply by code-asher** [reply]
Sometimes the bug is on our side but we would need to investigate further to be sure in this case (which is why we try to reproduce in Codespaces to see which side it is on).  But from what we know of the structure I think it seems likely to be on VS Code's side.

**Reply by AxelFlores1990** [reply]
I need something more concrete to open the issue from vscode side.. you say seems likely.. I understand that vscode doesen't have any web logic, so why it gonna be responsible of something related to browser operation?

**Reply by jsjoeio** [reply]
Did you try reproducing it in Codespaces? I think that'd be enough to open an issue

**Reply by AxelFlores1990** [reply]
To be honest with you i never use code spaces... you could't debug the process to check if its problem of vscode event propagation or if its not correctly handled by code server?

**Reply by jsjoeio** [reply]
Oh it's pretty straightforward to use: https://github.com/features/codespaces

No, but if you provide clear steps to reproduce and an extension to try it with, happy to try locally.

**Reply by AxelFlores1990** [reply]
Okas, Im going to try with codespaces to provide the error scenario
